libminifi/include/utils/StringUtils.h

```diff
@@ -432,6 +433,14 @@ struct ParseError {};
 nonstd::expected<std::optional<char>, ParseError> parseCharacter(std::string_view input);
 
 std::string replaceEscapedCharacters(std::string_view input);
+
+template<typename T>
+nonstd::expected<T, ParseError> parse(std::string_view input) {
+  T t{};
+  if (const auto result = std::from_chars(input.data(), input.data() + input.size(), t); result.ptr != input.data() + input.size())
+    return nonstd::make_unexpected(ParseError{});
+  return t;
+}
```

Comment:
Better naming and docs would be appreciated. From what I can see, this only parses numbers.
```suggestion
// no std::arithmetic yet
template <typename T> concept arithmetic = integral<T> || floating_point<T>;

template<arithmetic T>
nonstd::expected<T, ParseError> parseNumber(std::string_view input) {
  T t{};
  if (const auto result = std::from_chars(input.data(), input.data() + input.size(), t); result.ptr != input.data() + input.size())
    return nonstd::make_unexpected(ParseError{});
  return t;
}
```